contrib/golang/common/go/api/filter.go

```diff
@@ -210,3 +210,26 @@ type FilterState interface {
 	SetString(key, value string, stateType StateType, lifeSpan LifeSpan, streamSharing StreamSharing)
 	GetString(key string) string
 }
+
+type MetricType uint32
+
+const (
+	Counter   MetricType = 0
+	Gauge     MetricType = 1
+	Histogram MetricType = 2
+)
+
+type ConfigCallbacks interface {
+	// Define a metric, for different MetricType, name must be different,
+	// for same MetricType, the same name will share a metric.
+	DefineMetric(metricType MetricType, name string) Metric
+}
+
+type ConfigCallbackHandler interface {
+	ConfigCallbacks
+}
+
+type Metric interface {
```

Comment:
we'd better define different interfaces here, i.e. Counter, Gauge, Histogram. since they have different methods.
And we could reuse the Envoy C api between different Golang interface methods, if possible.

For this PR, implement Counter and Gauge is good enough, let's leave Histogram as TODO.

thoughts?